api/v1alpha1/selfnoderemediationconfig_webhook.go

```diff
@@ -0,0 +1,171 @@
+	"time"
+)
+
+const (
+	webhookCertDir  = "/apiserver.local.config/certificates"
+	webhookCertName = "apiserver.crt"
+	webhookKeyName  = "apiserver.key"
+)
+
+// fields names
+const (
+	PeerApiServerTimeout = "PeerApiServerTimeout"
+	ApiServerTimeout     = "ApiServerTimeout"
+	PeerDialTimeout      = "PeerDialTimeout"
+	PeerRequestTimeout   = "PeerRequestTimeout"
+	ApiCheckInterval     = "ApiCheckInterval"
+	PeerUpdateInterval   = "PeerUpdateInterval"
+)
+
+//minimal time durations allowed string format
+const (
+	MinDurPeerApiServerTimeout = "10ms"
+	MinDurApiServerTimeout     = "10ms"
+	MinDurPeerDialTimeout      = "10ms"
+	MinDurPeerRequestTimeout   = "10ms"
+	MinDurApiCheckInterval     = "1s"
+	MinDurPeerUpdateInterval   = "10s"
+)
+
+const (
+	ErrPeerApiServerTimeout = PeerApiServerTimeout + " can't be less than " + MinDurPeerApiServerTimeout
+	ErrApiServerTimeout     = ApiServerTimeout + " can't be less than " + MinDurApiServerTimeout
+	ErrPeerDialTimeout      = PeerDialTimeout + " can't be less than " + MinDurPeerDialTimeout
+	ErrPeerRequestTimeout   = PeerRequestTimeout + " can't be less than " + MinDurPeerRequestTimeout
+	ErrApiCheckInterval     = ApiCheckInterval + " can't be less than " + MinDurApiCheckInterval
+	ErrPeerUpdateInterval   = PeerUpdateInterval + " can't be less than " + MinDurPeerUpdateInterval
+)
+
+type field struct {
+	fieldDurValue       *metav1.Duration
+	fieldMinValueString string
+	errorMessage        string
+}
+
+// log is for logging in this package.
+var selfNodeRemediationConfigLog = logf.Log.WithName("selfnoderemediationconfig-resource")
+
+func (r *SelfNodeRemediationConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+
+	// check if OLM injected certs
+	certs := []string{filepath.Join(webhookCertDir, webhookCertName), filepath.Join(webhookCertDir, webhookKeyName)}
+	certsInjected := true
+	for _, fname := range certs {
+		if _, err := os.Stat(fname); err != nil {
+			certsInjected = false
+			break
+		}
+	}
+	if certsInjected {
+		server := mgr.GetWebhookServer()
+		server.CertDir = webhookCertDir
+		server.CertName = webhookCertName
+		server.KeyName = webhookKeyName
+	} else {
+		selfNodeRemediationConfigLog.Info("OLM injected certs for webhooks not found")
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-self-node-remediation-medik8s-io-v1alpha1-selfnoderemediationconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=self-node-remediation.medik8s.io,resources=selfnoderemediationconfigs,verbs=create;update,versions=v1alpha1,name=vselfnoderemediationconfig.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &SelfNodeRemediationConfig{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *SelfNodeRemediationConfig) ValidateCreate() error {
+	selfNodeRemediationConfigLog.Info("validate create", "name", r.Name)
+
+	return nil
```

Comment:
why do you check updates only, but no creates?

Reply:
I remember reading in the medik8s.io that users can't create the PPC CR, only edit it. So I figured only validating the update is a better way to prevent the problem that used to happen when the PPC object was created before the webhook server is started (now it's not a problem because the creation of the default PPC isn't validated). WDYT?

Reply:
Users can delete and recreate the config at any time. And also validating our own default config has value IMHO.